server-protocol.md

```diff
@@ -104,6 +117,50 @@ was sent in the wrong order), an `error` response will be sent, This response
 will include the error string in the `error` key, and a full copy of the
 original message dictionary in `orig`.
 
+
+## Permission to Use the Server
+
+Server operators may wish to deny service to some clients. We
+generally refer to this as "Permission" and imagine future additions
+of use-cases and methods, although only one such pair is decribed
+currently.
+
+One such use-case is if the server is under a Denial of Service (DoS)
+attack or other malicious activity. We describe a "proof of work"
+scheme for clients to gain permission to use a server under DoS
+attack.
+
+In the `welcome` message the server may include a `permission-required` key. If provided, this will point to a `dict` with one key per supported method along with any method-specific options.
+For example::
```

Comment:
```suggestion
For example:
```

Reply:
Isn't the double-colon is needed to render the JSON? Or am I thinking `.rst`?

Reply:
okay looks like I was remembering rST not markdown...

Reply:
Yes, for markdown you do:

```markdown
   ```json
   { "foo" = "bar" }
   \```
```

(Ignore the backslash, escaping in Markdown is weak)